diff --git a/commands/env.cmd b/commands/env.cmd
index a5ab22b..7f3227a 100644
--- a/commands/env.cmd
+++ b/commands/env.cmd
@@ -258,6 +258,30 @@ then
   fi
 fi
 
+if [[ "${WARDEN_PARAMS[0]}" == "up" ]] || [[ "${WARDEN_PARAMS[0]}" == "start" ]]; then
+    if [[ $($WARDEN_BIN env ps -q php-fpm) ]] && [ $($WARDEN_BIN env exec php-fpm "id" "-u" "www-data") != ${HOST_UID} ]; then
+        $WARDEN_BIN env exec -u 0 php-fpm "usermod" "-u" "${HOST_UID}" "www-data"
+        $WARDEN_BIN env exec -u 0 php-fpm "groupmod" "-g" "${HOST_GID}" "www-data"
+        $WARDEN_BIN env exec -u 0 php-fpm "groupadd" "-g" "1000" "centos"
+        $WARDEN_BIN env exec -u 0 php-fpm "usermod" "-aG" "centos" "www-data"
+        $WARDEN_BIN env restart php-fpm
+    fi
+    if [[ $($WARDEN_BIN env ps -q php-debug) ]] && [ $($WARDEN_BIN env exec php-debug "id" "-u" "www-data") != ${HOST_UID} ]; then
+        $WARDEN_BIN env exec -u 0 php-debug "usermod" "-u" "${HOST_UID}" "www-data"
+        $WARDEN_BIN env exec -u 0 php-debug "groupmod" "-g" "${HOST_GID}" "www-data"
+        $WARDEN_BIN env exec -u 0 php-debug "groupadd" "-g" "1000" "centos"
+        $WARDEN_BIN env exec -u 0 php-debug "usermod" "-aG" "centos" "www-data"
+        $WARDEN_BIN env restart php-debug
+    fi
+    if [[ $($WARDEN_BIN env ps -q php-spx) ]] && [ $($WARDEN_BIN env exec php-spx "id" "-u" "www-data") != ${HOST_UID} ]; then
+        $WARDEN_BIN env exec -u 0 php-spx "usermod" "-u" "${HOST_UID}" "www-data"
+        $WARDEN_BIN env exec -u 0 php-spx "groupmod" "-g" "${HOST_GID}" "www-data"
+        $WARDEN_BIN env exec -u 0 php-spx "groupadd" "-g" "1000" "centos"
+        $WARDEN_BIN env exec -u 0 php-spx "usermod" "-aG" "centos" "www-data"
+        $WARDEN_BIN env restart php-spx
+    fi
+fi
+
 ## stop mutagen sync if needed
 if [[ "${WARDEN_PARAMS[0]}" == "down" ]] \
     && [[ $OSTYPE =~ ^darwin ]] && [[ -f "${MUTAGEN_SYNC_FILE}" ]]
